src/utils/editor/source-search.js

```diff
@@ -157,6 +157,31 @@ function doSearch(ctx, rev, query, keepSelection, modifiers: SearchModifiers) {
   });
 }
 
+export function highlightQuery(
+  ctx: any,
+  rev: boolean,
+  query: string,
+  keepSelection: boolean,
+  modifiers: SearchModifiers,
+  line: number
+) {
+  const { cm } = ctx;
+  if (!cm) {
+    return;
+  }
+  const defaultIndex = { line: -1, ch: -1 };
+
+  return cm.operation(function() {
+    const state = getSearchState(cm, query);
+    const isNewQuery = state.query !== query;
+    state.query = query;
+
+    updateOverlay(cm, state, query, modifiers);
+    updateCursor(cm, state, keepSelection);
+    findNextOnLine(ctx, rev, query, isNewQuery, modifiers, line);
+  });
+}
```

Comment:
this looks similar to doSearch, are there are parts we want to share? Why do we need two functions?

Reply:
Yeah, the only differences are:
1. accepts `line` and `ch` so `findNextOnLine` can know which line to full highlight
2. calls `findNextOnLine` instead of `findNext`, which are pretty different right now